common/verity.c

```diff
@@ -150,9 +154,125 @@ verity_get_device_path_new(const char *label)
 	return mem_printf("%s%s", DM_PATH_PREFIX, label);
 }
 
+static int
+create_dm_symlink(const char *name, const dev_t devt, bool enforce_symlinks)
+{
+	int ret = -1;
+	uevent_event_t *uev = NULL;
+	char *targetpath = NULL, *linkpath = NULL;
+
+	DEBUG("Creating symlink for verity device %s (%lu:%lu)", name, MAJOR(devt), MINOR(devt));
+
+	char *uev_path = mem_printf("/sys/dev/block/%lu:%lu/uevent", MAJOR(devt), MINOR(devt));
+	char *buf = NULL;
+
+	if (!(buf = file_read_new(uev_path, PATH_MAX))) {
+		ERROR("Failed to read uevent for verity device %s", name);
+		goto out;
+	}
+
+	uev = uevent_parse_from_string(buf);
+
+	if (!uev) {
+		ERROR_ERRNO("Failed to parse uevent");
+		goto out;
+	}
+
+	const char *devptr = uevent_event_get_devname(uev);
+
+	if (!devptr) {
+		ERROR_ERRNO("Failed to parse devname from uevent");
+		mem_free0(uev);
+		goto out;
+	}
+
+	targetpath = mem_printf("%s/%s", DEVFS_PATH, devptr);
+	linkpath = mem_printf("%s/%s", DM_PATH_PREFIX, name);
+
+	DEBUG("Creating symlink for verity device %s: %s -> %s", name, linkpath, targetpath);
+
+	if (0 != symlink(targetpath, linkpath)) {
+		if ((errno == EEXIST) && (!enforce_symlinks)) {
+			WARN("Symlink for verity device %s already exists, using it as requested",
+			     name);
+		} else {
+			ERROR_ERRNO("Failed to create symlink for verity device %s: %s -> %s", name,
+				    linkpath, targetpath);
+			goto out;
+		}
+	}
+
+	ret = 0;
+
+out:
+	mem_free0(uev_path);
```

Comment:
remove all these newlines in cleanup code, plz

Reply:
done